### PR TITLE
feat: release theme designer as a beta product

### PR DESCRIPTION
**Pull request by SiTaggart**
- [x] convert to typescript
- [x] add an intro modal
- [x] Add Google analytics
- [x] Add Datadog RUM
- [x] Add link to doc site footer

### CODE REVIEW

File: packages/paste-theme-designer/pages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import * as React from 'react';
+import {EditorWrapper} from '../components/layout/editor-wrapper';
+
+export default function Home(): JSX.Element {
+  return <EditorWrapper>Select an category from the dropdown above to begin editing</EditorWrapper>;
```
Comment (serifluous) [inline packages/paste-theme-designer/pages/index.tsx:5]:
```suggestion
  return <EditorWrapper>Select a category above to begin editing.</EditorWrapper>;
```

File: packages/paste-theme-designer/netlify.toml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+[context.branch-deploy.environment]
+  ENVIRONMENT_CONTEXT = "branch-deploy"
+
+[context.production.environment]
+  ENVIRONMENT_CONTEXT = "production"
+
+[[headers]]
+  for = "/*"
+  [headers.values]
+    X-Shenanigans = "none"
```
Comment (TheSisb) [inline packages/paste-theme-designer/netlify.toml:18]: lol

File: packages/paste-theme-designer/utils/SimpleStorage.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+/** @module SimpleStorage */
+
+/** The namespace for all storage keys in SimpleStorage. */
+const BASE_KEY = 'paste-website';
```
Comment (TheSisb) [inline packages/paste-theme-designer/utils/SimpleStorage.ts:4]: change this to `theme-designer` or `paste-remix`

**Review by TheSisb — APPROVED** [review]
One small nit in SimpleStorage, the rest looks amazing

**Comment by TheSisb** [comment]
Should we rename the disk folder from `paste-theme-designer` to `paste-remix` to avoid future confusion?

File: packages/paste-theme-designer/components/topbar/SiteHeaderLogo.tsx [hunk 4]
```diff
@@ -34,20 +40,23 @@ const SiteHeaderLogo = ({title, subtitle}) => {
         </MediaFigure>
```
Comment (nkrantz) [inline packages/paste-theme-designer/components/topbar/SiteHeaderLogo.tsx:40]: The Paste logo feels a little bare not being in a box. Can we use the red background one instead?

Comment (SiTaggart) [reply packages/paste-theme-designer/components/topbar/SiteHeaderLogo.tsx:40]: It's the one we used to have on the docsite before the pride logo I believe.

File: packages/paste-theme-designer/components/topbar/BetaModal.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+import {useUID} from '@twilio-paste/core/uid-library';
+import {Modal, ModalHeader, ModalHeading, ModalBody} from '@twilio-paste/core/modal';
+import {AspectRatio} from '@twilio-paste/core/aspect-ratio';
+import {Paragraph} from '@twilio-paste/core/paragraph';
+import type {ModalProps} from '@twilio-paste/core/modal';
+import {Anchor} from '@twilio-paste/core/anchor';
+
+export type BetaModalProps = Pick<ModalProps, 'isOpen' | 'onDismiss'>;
+
+const BetaModal: React.FC<BetaModalProps> = ({isOpen, onDismiss}) => {
```
[the remark this answers is not in the document]
Comment (SiTaggart) [reply packages/paste-theme-designer/components/topbar/BetaModal.tsx:11]: Resolved from copy update

File: packages/paste-theme-designer/components/topbar/BetaModal.tsx [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  const modalID = useUID();
+  return (
+    <Modal isOpen={isOpen} ariaLabelledby={modalID} onDismiss={onDismiss} size="wide">
+      <ModalHeader>
+        <ModalHeading id={modalID}>Welcome to Paste Remix</ModalHeading>
+      </ModalHeader>
+      <ModalBody>
+        <Paragraph>
+          Paste Remix is an interactive design tool aimed at creating custom themes for applications built using Paste.
```
[the remark this answers is not in the document]
Comment (SiTaggart) [reply packages/paste-theme-designer/components/topbar/BetaModal.tsx]: Resolved with copy update

File: packages/paste-theme-designer/components/topbar/SiteHeaderLogo.tsx (hunk 4 above)
Comment (nkrantz) [reply packages/paste-theme-designer/components/topbar/SiteHeaderLogo.tsx:40]: I thought it used to be a white logo in a red square box ? I could be remembering wrong

**Pull request merged**